File: src/main/java/com/github/joelws/release/tracker/dao/release/ReleaseDaoImpl.java
package com.github.joelws.release.tracker.dao.release;

import com.github.joelws.release.tracker.dao.GenericDaoImpl;
import com.github.joelws.release.tracker.entity.release.Release;

import org.springframework.stereotype.Repository;

import java.util.List;

@Repository
public class ReleaseDaoImpl extends GenericDaoImpl<Release, String> implements ReleaseDao {

    public ReleaseDaoImpl() {
        super();
    }

    @SuppressWarnings("unchecked")
    @Override
    public List<Release> list() {
        final String query = "SELECT r FROM Release r";
        return (List<Release>) getEntityManager().createQuery(query).getResultList();
    }
}

File: src/main/java/com/github/joelws/release/tracker/dao/release/ReleaseDaoImpl.kt
package com.github.joelws.release.tracker.dao.release

import com.github.joelws.release.tracker.dao.GenericDaoImpl
import com.github.joelws.release.tracker.entity.release.Release

import org.springframework.stereotype.Repository

@Repository
open class ReleaseDaoImpl : GenericDaoImpl<Release, String>(), ReleaseDao {

    @SuppressWarnings("unchecked")
    override fun list(): List<Release> {
        val query = "SELECT r FROM Release r"
        return entityManager!!.createQuery(query).resultList as List<Release>
    }
}
